### PR TITLE
add -Wshadow and fix all occurrences

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Review by rfomin — APPROVED** [review]

File: src/m_cheat.c [hunk 1]
```diff
@@ -1012,10 +1012,10 @@ boolean M_FindCheats(int key)
           const char *p; // [FG] char!
           for (p=cheat[i].cheat; *p; p++)
             {
-              unsigned key = tolower(*p)-'a';  // convert to 0-31
```
Comment (kraflab) [inline src/m_cheat.c:1015]: What's the purpose of these changes? In this case for instance we've moved from something with semantic clarity (key) into something that has no meaning at all (low).

Comment (rfomin) [reply src/m_cheat.c:1015]: `key` is a function parameter in `M_FindCheats()`, so we "shadow" it here, see the description of the `-Wshadow` warning: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

> into something that has no meaning at all (low).

Well, that means the result of the `tolower()` function. Sure, we can come up with better names, but I think the Doom codebase is hopeless in this aspect.😄

Comment (kraflab) [reply src/m_cheat.c:1015]: The code base overall may be rough but we can still make incremental improvements. I'd prefer it if the fixes for this warning rather improved the clarity of the variables, otherwise I think it's missing some of the reason to use it. For instance instead of switching from "i" to "t" maybe something should be "line_index". Just an example. And of course it's not my codebase, just my opinion :^)

Comment (fabiangreffrath) [reply src/m_cheat.c:1015]: While I was messing up with the widgets drawing implementation in `hu_stuff.c` I learned the hard way that I use *way* too generic names for my enums and variables. Thus, I decided to add the `-Wshadow` warning and was surprised how many occurrences it found.

I am not even sure if I will merge this PR. And single mistake and it will mix up global and scoped variables, which may lead to the most random desync in the most obscure demo and the cause will likely never be found. After all, it was a learning session for me to see how "rough" the code base is in this regard.

Regarding the renaming of variables, of course there are always better names to come up with. But in this specific case, calling `tolower()` on a key turns this into the "low" key. Still sub optimal, I know. :grin:

Comment (rfomin) [reply src/m_cheat.c:1015]: `unsigned key = tolower(*p)-'a';  // convert to 0-31` there is also `a` subtraction, so it's not a "low" key, maybe we should call it `num`?

I think we can use this as an excuse for better naming/formatting.

**Comment by fabiangreffrath** [comment]
Not convinced of my own idea.
